content/blog/2019/corda-responder-validation/index.md

```diff
@@ -0,0 +1,185 @@
+---
+title: Responder flow validation
+date: "2019-11-25"
+published: true
+tags: [corda, kotlin, dlt, distributed ledger technology, blockchain]
+cover_image: ./title-card.png
+---
+
+Do you want to prevent your node from accepting transactions that are not important to your business?
+
+Do you want to prevent your node from accepting transactions that are invalid from your business's perspective?
+
+If you answered yes to either of these questions. Then you might want to consider adding some validation to your responder flows.
+
+For context, a responder flow is executed by a counterparty node when communicating with an initiating flow. Part of a responder flow's duty typically includes signing and recording a transaction sent to it. Meaning an organisation can end up in a situation where they did not have a large involvement in putting the contents of the transaction together, yet are expected to sign and potentially record it.
+
+Surely, there should be some rules to prevent an organisation from being sent any old rubbish? And, yes, there are. The very bare minimum that needs to pass is contract validation. This ensures that a transaction is logically valid at a general level. What it does not do. Is enforce local identity/roles or business level requirements upon a transaction.
+
+To an extent, this is fine. A __contract should__ only be ensuring that a transaction is __logically valid__.
```

Comment:
Move the emphasis off contract, keep it on should, or remove it from both, either is fine.